pkg/segment/results/segresults/segresults.go

```diff
@@ -103,6 +104,24 @@ type segStatsResults struct {
 	groupByCols      []string
 }
 
+type RemoteStats struct {
+	EvalStats map[string]EvalStatsMetaData
+	SegStats  []*structs.SegStats
+}
+
+type EvalStatsMetaData struct {
+	RangeStat     *structs.RangeStat
+	AvgStat       *structs.AvgStat
+	StrSet        map[string]struct{}
+	StrList       []string
+	MeasureResult interface{} // we should not use CValueEnclosure directly, because it treats interface having numbers as float64
```

Comment:
It has the SS_DTYPE but since CVal is interface when we marshal and unmarshal the CValEnclosure the CVal interface will have the numeric value but the data type would be changed to float64 for numeric values.

Reply:
This design for sending data is also used over [here](https://github.com/siglens/siglens/blob/e8aa3e5de775da20e31273011352190a8627ddd2/pkg/segment/results/blockresults/blockresult.go#L957).

Reply:
addressed in the latest commit.